pkg/servicemanager/startuprecorder/startuprecorder.go

```diff
@@ -0,0 +1,71 @@
+package startuprecorder
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+type ServiceData struct {
+	Name         string        `json:"name"`
+	Dependencies []string      `json:"dependencies"`
+	Start        time.Time     `json:"start"`
+	Ready        time.Time     `json:"ready"`
+	TimeToReady  time.Duration `json:"timeToReady"`
+}
+
+type StartupRecorder struct {
+	Services              []ServiceData `json:"services"`
+	MicroshiftStart       time.Time     `json:"microshiftStart"`
+	MicroshiftReady       time.Time     `json:"microshiftReady"`
+	MicroshiftTimeToReady time.Duration `json:"microshiftTimeToReady"`
+
+	m sync.Mutex
```

Comment:
Looking at these members I wonder if everything except mutex should be its own structure, so we don't mix the "Recorder" and the data it gathers. It looks a bit off when Recorder marshals itself to json.

Reply:
Added a StartupData struct to separate the recorder and the data